frontend/controller/router.js

```diff
@@ -70,113 +68,133 @@ function createEnterGuards (...guards) {
     next()
   }
 }
+const routes = [
+  {
+    path: '/',
+    component: Home,
+    name: 'home',
+    meta: { title: L('Group Income') }, // page title. see issue #45
+    beforeEnter: createEnterGuards(homeGuard)
+  },
+  {
+    path: '/design-system',
+    component: lazyDesignSystem,
+    name: 'DesignSystem',
+    meta: { title: L('Design System') }
+    // beforeEnter: createEnterGuards(designGuard)
+  },
+  {
+    path: '/dashboard',
+    component: lazyGroupDashboard,
+    name: 'GroupDashboard',
+    meta: { title: L('Group Dashboard') },
+    beforeEnter: createEnterGuards(loginGuard, groupGuard)
+  },
+  {
+    path: '/contributions',
+    component: lazyContributions,
+    name: 'Contributions',
+    meta: { title: L('Contributions') },
+    beforeEnter: createEnterGuards(loginGuard, groupGuard)
+  },
+  {
+    path: '/payments',
+    component: lazyPayments,
+    name: 'Payments',
+    meta: { title: L('Payments') },
+    beforeEnter: createEnterGuards(loginGuard, groupGuard)
+  },
+  /* Guards need to be created for any route that should not be directly accessed by url */
+  {
+    path: '/messages',
+    component: lazyMessages,
+    name: 'Messages',
+    meta: { title: L('Messages') },
+    beforeEnter: createEnterGuards(loginGuard)
+  },
+  {
+    path: '/messages/:chatName',
+    component: lazyMessages,
+    name: 'MessagesConversation',
+    beforeEnter: createEnterGuards(loginGuard)
+    // BUG/REVIEW "CANNOT GET /:username" when username has "." in it
+    // ex: messages/joe.kim doesnt work but messages/joekim works fine.
+    // Possible Solution: https://router.vuejs.org/guide/essentials/history-mode.html#example-server-configurations
+  },
+  {
+    path: '/group-chat',
+    component: lazyGroupChat,
+    name: 'GroupChat',
+    meta: { title: L('Group Chat') },
+    beforeEnter: createEnterGuards(loginGuard, groupGuard)
+  },
+  {
+    path: '/group-settings',
+    component: lazyGroupSettings,
+    name: 'GroupSettings',
+    meta: { title: L('Group Settings') },
+    beforeEnter: createEnterGuards(loginGuard, groupGuard)
+  },
+  {
+    path: '/group-chat/:chatRoomId',
+    component: lazyGroupChat,
+    name: 'GroupChatConversation',
+    beforeEnter: createEnterGuards(loginGuard, groupGuard)
+  },
+  {
+    path: '/join',
+    name: Join.name,
+    component: Join,
+    meta: { title: L('Join a Group') },
+    // beforeEnter: createEnterGuards(loginGuard, mailGuard)
+    beforeEnter: createEnterGuards(inviteGuard)
+  },
+  ...(process.env.NODE_ENV === 'development'
+    ? [{
+        path: '/error-testing',
+        name: 'ErrorTesting',
+        component: () => import('../views/pages/ErrorTesting.vue'),
+        meta: { title: L('Error Testing') }
+      }]
+    : []),
+  {
+    path: '*',
+    redirect: '/'
+  }
+]
 const router: any = new Router({
   mode: 'history',
   base: '/app',
+  routes,
```

Comment:
Why was `routes` extracted?

Reply:
This should be answered a few lines down bellow...